concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -367,6 +381,85 @@ exportDatabaseV3 dbDir outDir chunkSize = do
+exportConsensusV1Blocks ::
+    forall pv m.
+    ( IsProtocolVersion pv,
+      MonadIO m,
+      KonsensusV1.MonadTreeStateStore m
+    ) =>
+    -- |Export path.
+    FilePath ->
+    -- |Chunk size
+    Word64 ->
+    -- |The genesis index.
+    GenesisIndex ->
+    -- |Height to start export from.
+    BlockHeight ->
+    -- |The block index of the previous export.
+    BlockIndex ->
+    -- |Last written chunk in previous export
+    Maybe String ->
+    -- |Returns a @Bool@ which indicates whether anything went wrong,
+    -- i.e. it is 'True' if an error occured and otherwise 'False,
+    -- and the resulting 'BlockIndex'.
+    m (Bool, BlockIndex)
+exportConsensusV1Blocks outDir chunkSize genIndex startHeight blockIndex lastWrittenChunkM = do
+    KonsensusV1.lookupFirstBlock >>= \case
+        Nothing -> do
+            liftIO . putStrLn $ "Could not read from database."
+            return (True, Empty)
+        Just genesisBlock -> do
+            let exportedGenHash = exportedGenHashOr genHash blockIndex
+                genHash = getHash genesisBlock
+            if genHash /= exportedGenHash
+                then do
+                    liftIO . putStrLn $ "Genesis hash does not match the recently exported block index."
+                    return (True, Empty)
+                else do
+                    KonsensusV1.lookupLastBlock >>= \case
+                        Nothing -> do
+                            liftIO . putStrLn $ "Cannot read last block of the database."
+                            return (True, Empty)
+                        Just sb -> do
+                            let getBlockAt height =
+                                    KonsensusV1.lookupBlockByHeight height >>= \case
+                                        Nothing -> return Nothing
+                                        Just b | KonsensusV1.NormalBlock signedBlock <- KonsensusV1.stbBlock b -> do
+                                            let serializedBlock = runPut $ KonsensusV1.putSignedBlock signedBlock
+                                            return $ Just (serializedBlock, Nothing)
+                                        _ -> return Nothing -- Do not export genesis blocks.
+                                getFinalizationRecordAt :: FinalizationIndex -> m (Maybe BS.ByteString)
+                                getFinalizationRecordAt _ = return Nothing
+                            chunks <-
+                                writeChunks
+                                    genIndex
+                                    (demoteProtocolVersion (protocolVersion @pv))
+                                    genHash
+                                    startHeight
+                                    (KonsensusV1.bmHeight . KonsensusV1.stbInfo $ sb)
+                                    outDir
+                                    chunkSize
+                                    lastWrittenChunkM
+                                    0
```

Comment:
So this 0 here is a dummy value, right? Can we document that please?

Reply:
Yes. Documented.